pkg/util/util.go

```diff
@@ -232,48 +231,27 @@ func ExecuteWithTimeout(timeout time.Duration, envs []string, binary string, arg
 	return output.String(), nil
 }
 
-func DetectFileFormat(filePath string) (string, error) {
-
-	/* Example command outputs
-	   $ qemu-img info parrot.raw
-	   image: parrot.raw
-	   file format: raw
-	   virtual size: 32M (33554432 bytes)
-	   disk size: 2.2M
-
-	   $ qemu-img info parrot.qcow2
-	   image: parrot.qcow2
-	   file format: qcow2
-	   virtual size: 32M (33554432 bytes)
-	   disk size: 2.3M
-	   cluster_size: 65536
-	   Format specific information:
-	       compat: 1.1
-	       lazy refcounts: false
-	       refcount bits: 16
-	       corrupt: false
-	*/
-
-	output, err := Execute([]string{}, QemuImgBinary, "info", filePath)
-	if err != nil {
-		return "", err
-	}
+type QemuImgInfo struct {
+	// For qcow2 files, VirtualSize may be larger than the physical
+	// image size on disk.  For raw files, `qemu-img info` will report
+	// VirtualSize as being the same as the physical file size.
+	VirtualSize int64  `json:"virtual-size"`
+	Format      string `json:"format"`
+}
 
-	scanner := bufio.NewScanner(strings.NewReader(output))
-	for scanner.Scan() {
-		line := strings.TrimSpace(scanner.Text())
-		if strings.HasPrefix(line, "file format: ") {
-			return strings.TrimPrefix(line, "file format: "), nil
-		}
+func GetQemuImgInfo(filePath string) (imgInfo QemuImgInfo, err error) {
+	output, err := Execute([]string{}, QemuImgBinary, "info", "--output=json", filePath)
```

Comment:
NIT: It's nice to provide an example output as the comment

Reply:
Done!